addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1581,6 +1582,442 @@ var FieldBinaryImage = AbstractFieldBinary.extend({
+     * Render and scale the image in the canvas.
+     * @param {string} imgSrc image (either a generated signature or a uploaded image)
+     */
+    _printImage: function (imgSrc) {
+        var self = this;
+
+        var image = new Image;
+        image.onload = function () {
+            var width = 0, height = 0;
+            var ratio = image.width / image.height;
+
+            self.$signatureField.jSignature('reset');
+            var $canvas = self.$signatureField.find('canvas');
+            var context = $canvas[0].getContext("2d");
+
+            if (image.width / $canvas[0].width > image.height / $canvas[0].height) {
+                width = $canvas[0].width;
+                height = width / ratio;
+            } else {
+                height = $canvas[0].height;
+                width = height * ratio;
+            }
+
+            setTimeout(function () {
+                var ignoredContext = _.pick(context, ['shadowOffsetX', 'shadowOffsetY']);
+                _.extend(context, {shadowOffsetX: 0, shadowOffsetY: 0});
+                context.drawImage(image, 0, 0, image.width, image.height, ($canvas[0].width - width) / 2, ($canvas[0].height - height) / 2, width, height);
+                _.extend(context, ignoredContext);
+            }, 0);
+        };
+        image.src = imgSrc;
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * Select a signature mode: draw manually, automatic generation or upload a signature image
+     * @param {MouseEvent} e
+     */
+    _onClickChangeSignatureMode: function (e) {
+        this.$modeButtons.removeClass('btn-primary');
+        $(e.target).addClass('btn-primary');
+
+        this.mode = $(e.target).data('mode');
+
+        if (SignatureDialog.fonts.length > 1) {
+            this.$selectStyleButton.toggle(this.mode === 'auto');
+        }
+        this.$clearButton.toggle(this.mode === 'draw');
+        this.$loadButton.toggle(this.mode === 'load');
+
+        if (this.mode === 'load') {
+            this.$loadButton.click();
+        } else if (this.mode === 'auto') {
+            this._printText(SignatureDialog.fonts[this.currentFont], this._getSignatureText());
+        } else {
+            $('.o_sign_load').val("");
+        }
+
+        this.$signatureField.jSignature((this.mode === 'draw') ? "enable" : "disable");
+
+        this.$fontDialog.hide().css('width', 0);
+    },
+
+    /**
+     * @private
+     * Open selector with different fonts for automatic generation
+     */
+    _onClickOpenFont: function () {
+        var self = this;
+        this.$fontDialog.find('a').empty().append($('<div/>'));
+        this.$fontDialog.show().animate({'width': self.$fontDialog.find('a').first().height() * self.signatureRatio * 1.25}, 500, function () {
+            self.$fontDialog.find('a').each(function (i, el) {
```

Comment:
Note that this function will only be called after the animation is completed, so after 500 ms (and not during). Is it the intention ?